### PR TITLE
Update supported mmseg models

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Test deployment of mmseg models.

## Modification

- [x] Test all the deployment of all mmseg models. Totally 19 new models are supported in mmseg. Now 25/33 models are supported in mmseg. 
- [x] Add rewriting for EMAModule to replace `torch.einsum` op and its unit test.
- [x] Add `MAX_TOPK_K` for rewriting of TopK with `topk__tensorrt`
- [x] Updated docs.

#### Note 
- for models with `slide` inference mode, we test with static shape and directly resize image to a shape such as `1024x1024`
- DANet uses too much MEM(>9G). Could not test on dataset with 1024x2048.

| Model                       | PyTorch(Slide) | PyTorch(Whole) | TensorRT(Whole) |    Model config |                               
|:----------------------------|:-----------:|:--------:|:----:|:----------------------------------------------------------------------------------------:|
|Segmenter | 45.75(crop-512x512)| 44.32(resize-512x512)| 44.29(resize-512x512)| configs/segmenter_vit-s_linear_8x1_512x512_160k_ade20k.py|
|Segformer| 76.54(crop-1024x1024)| 68.36(resize-1024x1024)| 68.34(resize-1024x1024)| configs/segmenter_vit-s_linear_8x1_512x512_160k_ade20k.py|

## BC-breaking (Optional)

None

## Use cases (Optional)

Test all models with static shape.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RunningLeon** [comment]
@SemyonBevzuk Hi, could you kindly check the models that fail on openvino in this table： https://github.com/open-mmlab/mmdeploy/blob/4bb98bdf125fd5d185e1b94f16d0a8b5bd438d44/docs/en/codebases/mmseg.md
Thanks a lot.

**Comment by SemyonBevzuk** [comment]
> @SemyonBevzuk Hi, could you kindly check the models that fail on openvino in this table： https://github.com/open-mmlab/mmdeploy/blob/4bb98bdf125fd5d185e1b94f16d0a8b5bd438d44/docs/en/codebases/mmseg.md Thanks a lot.

Hi! I will check what can be done.

**Comment by RunningLeon** [comment]
> > @SemyonBevzuk Hi, could you kindly check the models that fail on openvino in this table： https://github.com/open-mmlab/mmdeploy/blob/4bb98bdf125fd5d185e1b94f16d0a8b5bd438d44/docs/en/codebases/mmseg.md Thanks a lot.
> 
> Hi! I will check what can be done.

Thanks a lot

File: configs/mmseg/segmentation_openvino_static-512x512.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+_base_ = ['./segmentation_static.py', '../_base_/backends/openvino.py']
+
+backend_config = dict(
+    model_inputs=[dict(opt_shapes=dict(input=[1, 3, 512, 512]))])
```
Comment (AllentDan) [inline configs/mmseg/segmentation_openvino_static-512x512.py:4]: As far as I know, OpenVINO in `mmdeploy` seems only support dynamic? @SemyonBevzuk

Comment (RunningLeon) [reply configs/mmseg/segmentation_openvino_static-512x512.py:4]: well this config is tested and no error happens.

Comment (AllentDan) [reply configs/mmseg/segmentation_openvino_static-512x512.py:4]: Yeah... But just like PPL.NN, the framework itself has no attribute of `static`, shall we provide static configs to users then?

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.

File: configs/mmseg/segmentation_openvino_static-512x512.py (hunk 1 above)
Comment (AllentDan) [reply configs/mmseg/segmentation_openvino_static-512x512.py:4]: Not a big deal if we keep static configs then

**Review by AllentDan — APPROVED** [review]
LGTM

File: mmdeploy/codebase/mmseg/models/decode_heads/ema_head.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+from mmdeploy.core import FUNCTION_REWRITER
+
+
+@FUNCTION_REWRITER.register_rewriter(
+    func_name='mmseg.models.decode_heads.ema_head.EMAModule.forward')
+def ema_module__forward(ctx, self, feats):
+    """Rewrite `forward` for default backend.
+
+    Replace torch.einsum with other operations.
```
Comment (AllentDan) [inline mmdeploy/codebase/mmseg/models/decode_heads/ema_head.py:13]: Shall we just rewrite torch.einsum rather than mmseg codes?

Comment (RunningLeon) [reply mmdeploy/codebase/mmseg/models/decode_heads/ema_head.py:13]: `torch.einsum` is a bit complex when considering all cases. Could be handled in another PR.

**Review by AllentDan — APPROVED** [review]

**Comment by RunningLeon** [comment]
@hanrui1sensetime Any comment on this PR?

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.

**Pull request merged**